android/src/main/java/com/cuhacking/atlas/search/SearchViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package com.cuhacking.atlas.search
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.cuhacking.atlas.db.Feature_fts
+import com.cuhacking.atlas.db.database
+
+class SearchViewModel : ViewModel() {
+
+    private val _searchResults: MutableLiveData<List<SearchResult>> = MutableLiveData()
+    val searchResults: LiveData<List<SearchResult>> = _searchResults
+
+        fun getSearchResults(query: String) {
+        val results: List<Feature_fts> = database.featureQueries.search(query).executeAsList()
+
+        val searchResultItems = results.map {
+            // Sample Search Result data (could be changed later)
+            SearchResult(it.common_name, it.secondary_name.toString() + ", " + it.type)
+        }
+        _searchResults.value = searchResultItems
+    }
+}
```

Comment:
For this project we'll also have our ViewModels in the `common` module since they can be reused across all platforms. As a result, we won't actually use the `ViewModel` class from the AndroidX library.

This has some technical implications... but we won't deal with those until we have a stronger footing on the Android side of things.

Reply:
Moving it to the `common` module will require some changes to the way `searchResults` is represented.

Instead of using `LiveData`, you can use [`StateFlow`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-state-flow/) which is part of Kotlin coroutines (and is already included in the project). For all intents and purposes right now, they behave just like `LiveData`, but is available in common code. 

Observing it in the Activity will be a bit different because of how `Flow` (and mainly coroutines in general) works. You'll have to `collect` (instead of `observe`) it like this:

```kotlin
viewLifecycleOwner.lifecycleScope.launch {
  myFlow.collect { list ->
    // Do something with the list
  }
}
```